Lib/test/test_typing.py

```diff
@@ -4265,11 +4268,20 @@ def test_valid_uses(self):
         self.assertEqual(C4.__args__, (P, T))
         self.assertEqual(C4.__parameters__, (P, T))
 
-        # ParamSpec instances should also have args and kwargs attributes.
+    def test_args_kwargs(self):
+        P = ParamSpec('P')
         self.assertIn('args', dir(P))
         self.assertIn('kwargs', dir(P))
-        P.args
-        P.kwargs
+        self.assertIsInstance(P.args, ParamSpecArgs)
+        self.assertIsInstance(P.kwargs, ParamSpecKwargs)
+        self.assertIs(P.args.__origin__, P)
+        self.assertIs(P.kwargs.__origin__, P)
+        self.assertEqual(repr(P.args), "~P.args")
+        self.assertEqual(repr(P.kwargs), "~P.kwargs")
+
+        P_co = ParamSpec("P_co", covariant=True)
+        self.assertEqual(repr(P_co.args), "+P_co.args")
+        self.assertEqual(repr(P_co.kwargs), "+P_co.kwargs")
```

Comment:
Oh, I didn't realize `ParamSpec` uses `+` to show it's covariant. It's clever, although a bit of an innovation for an edge case. It suddenly strikes me as slightly inappropriate here, since this is parsed as `+(P_co.args)` whereas the meaning is closer to `(+P_co).args`. But I wouldn't want to introduce those parentheses. Perhaps the variance of the `args` or `kwargs` attribute doesn't matter, and the repr() should not include the `~`, `+` or `-` sign? I don't think this applies for `TypeVar`, since it has no attributes.

Reply:
It does look a bit weird here, I'll change the repr()s to include only the name. The variance of ParamSpecs has no defined meaning anyway, and even if it did it's not essential for it to be reflected here.